vsintegration/src/FSharp.Editor/Navigation/NavigateToSearchService.fs

```diff
@@ -250,14 +254,18 @@ type internal FSharpNavigateToSearchService
                         |> Array.filter (fun x -> x.Name.Length = 1 && String.Equals(x.Name, searchPattern, StringComparison.InvariantCultureIgnoreCase))
                     else
                         [| yield! items |> Array.map (fun items -> items.Find(searchPattern)) |> Array.concat
-                           use patternMatcher = new PatternMatcher(searchPattern, allowFuzzyMatching = true)
+                           let patternMatcherOptions = PatternMatcherCreationOptions(
+                               cultureInfo = CultureInfo.CurrentUICulture,
+                               flags = PatternMatcherCreationFlags.AllowFuzzyMatching
+                           )
+                           let patternMatcher = patternMatcherFactory.CreatePatternMatcher(searchPattern, patternMatcherOptions)
                            yield! items
                                   |> Array.collect (fun item -> item.AllItems)
-                                  |> Array.Parallel.collect (fun x -> 
```

Comment:
Cancellation does not propagate to Array.Parallel. We should use Async.Parallel instead.